src/Build/Logging/FancyLogger/ANSIBuilder.cs

```diff
@@ -14,6 +15,35 @@ public static string ANSIRemove(string text)
             return Regex.Replace(text, "\\x1b(?:[@-Z\\-_]|\\[[0-?]*[ -\\/]*[@-~])", "");
         }
 
+        public static int ANSIBreakpoint(string text, int position)
+        {
+            if (position >= text.Length) return text.Length;
+            // Get substring
+            string substring = text.Substring(0, position);
+            string substringWithoutANSI = ANSIRemove(substring);
+            // Get length difference
+            int difference = substring.Length - substringWithoutANSI.Length;
```

Comment:
This would ignore any ansi characters between position and position + difference, right?

Reply:
Yes but as you said it does not consider the edge case where there is an ANSI string being clipped afterwards. Newest commit should fix that